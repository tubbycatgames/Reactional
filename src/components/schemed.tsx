import React, { ComponentType } from 'react'
import { Text, View } from 'react-native'
import { useColorScheme } from 'react-native-appearance'

import { getBackgroundStyle, getPrimaryTextStyle } from './colors'

export const withSchemedStyle = (
  Wrapped: ComponentType,
  getStyle: Function
) => {
  return (props: any) => (
    <Wrapped {...props} style={[getStyle(useColorScheme()), props.style]} />
  )
}

export const SchemeText = withSchemedStyle(Text, getPrimaryTextStyle)
export const SchemeBackground = withSchemedStyle(View, getBackgroundStyle)
