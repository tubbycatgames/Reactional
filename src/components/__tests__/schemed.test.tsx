import React from 'react'
import renderer from 'react-test-renderer'
import { View } from 'react-native'

import { withSchemedNav, withSchemedStyle } from '../schemed'

describe('withSchemedStyle', () => {
  it('should wrap a component with a scheme driven style', () => {
    const getStyle = () => ({ backgroundColor: 'transparent' })
    const SchemedView = withSchemedStyle(View, getStyle)

    const render = renderer.create(
      <SchemedView style={{ color: 'blue' }} testId='A Test' />
    )
    expect(render.toJSON()).toMatchSnapshot()
  })
})

describe('withSchemedNav', () => {
  it('should wrap a nav stack with scheming params', () => {
    const SchemedNav = withSchemedNav(({ screenOptions }) => {
      expect(screenOptions).toMatchSnapshot()
      return <View />
    })
    const render = renderer.create(<SchemedNav />)
    expect(render.toJSON()).toMatchSnapshot()
  })
})
