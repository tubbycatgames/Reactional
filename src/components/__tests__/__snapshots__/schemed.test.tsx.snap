// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSchemedStyle should wrap a component with a scheme driven style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "color": "blue",
      },
    ]
  }
  testId="A Test"
/>
`;
